main.py

```diff
@@ -39,39 +39,24 @@
 
 """
 
-from typing import (
-    List,
-    TYPE_CHECKING,
-    Tuple,
-    Pattern,
-    Dict,
-    Any,
-    Optional,
-    Union,
-    cast,
-)
-
-import sys
+import logging
 import os
-import time
 import re
-import logging
+import sys
+import time
 from datetime import datetime
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Pattern, Tuple, Union, cast
```

Comment:
Hmm, it is canonical to have imports from typing first in the import sequence...